src/v2/Components/Search/SearchBar.tsx

```diff
@@ -158,11 +160,29 @@ export class SearchBar extends Component<Props, State> {
     this.throttledFetch = throttle(this.throttledFetch, 500, {
       leading: true,
     })
+
     this.throttledOnSuggestionHighlighted = throttle(
       this.throttledOnSuggestionHighlighted,
       500,
       { leading: true }
     )
+
+    // Clear the search term once you navigate away from search results
+    this.removeNavigationListener = this.props.router
+      ? this.props.router.addNavigationListener(location => {
+          if (!location.pathname.startsWith("/search")) {
+            this.setState({ term: "" })
+          }
+
+          return true
+        })
+      : () => {
+          // noop
+        }
+  }
+
+  componentWillUnmount() {
+    this.removeNavigationListener()
```

Comment:
@damassi I think this captures the original intent behind clearing the input? I'm not 100% clear on how to test this, do we need to just wrap everything in a mock router etc? But it's pretty straightforward, regardless.